pkg/cmd/grafana/apiserver/cmd.go

```diff
@@ -116,16 +117,17 @@ type lateInitializedTracingProvider struct {
 }
 
 func (tp lateInitializedTracingProvider) Tracer(name string, options ...trace.TracerOption) trace.Tracer {
-	return tp.tracer
```

Comment:
This was the main problem. In general just changing this to `tp.tracer.Tracer` resolved the stack overflow panic.